llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -9352,6 +9352,27 @@ SDValue TargetLowering::expandAVG(SDNode *N, SelectionDAG &DAG) const {
     }
   }
 
+  if (VT.isScalarInteger() && !isTypeLegal(VT)) {
+    SDValue UAddWithOverflow = DAG.getNode(ISD::UADDO, dl, 
+                                           DAG.getVTList(VT, MVT::i1), { RHS, LHS });
+
+    SDValue Sum = UAddWithOverflow.getValue(0);
+    SDValue Overflow = UAddWithOverflow.getValue(1);
+
+    // Right shift the sum by 1
+    SDValue One = DAG.getShiftAmountConstant(1, VT, dl);
+    SDValue LShrVal = DAG.getNode(ISD::SRL, dl, VT, Sum, One);
+    
+    // Creating the select instruction
+    SDValue ZeroOut = DAG.getConstant(0, dl, VT); 
+    SDValue ZeroExtOverflow = DAG.getNode(ISD::ZERO_EXTEND, dl, VT, Overflow);
+    SDValue OverflowShl = DAG.getNode(ISD::SHL, dl, VT, ZeroExtOverflow, 
+                                      DAG.getConstant(VT.getScalarSizeInBits() - 1, dl, VT)); 
+    SDValue SelectVal = DAG.getSelect(dl, VT, Overflow, OverflowShl, ZeroOut);
+
+    return DAG.getNode(ISD::OR, dl, VT, LShrVal, SelectVal);
```

Comment:
You don't need to do the select anymore - you can simplify this to:
```cpp
SDValue ZeroExtOverflow = DAG.getNode(ISD::ZERO_EXTEND, dl, VT, Overflow);
SDValue OverflowShl = DAG.getNode(ISD::SHL, dl, VT, ZeroExtOverflow,  DAG.getConstant(VT.getScalarSizeInBits() - 1, dl, VT)); 
return DAG.getNode(ISD::OR, dl, VT, LShrVal, OverflowShl);
```

Reply:
I see, that makes more sense. I'll make sure to change that. Would i need to regenerate the test files over again?

Reply:
Yes, if you run `ninja check-llvm-codegen` it will tell you the failures, then use the update_llc_test_checks.py script to regenerate the files

Reply:
> Yes, if you run `ninja check-llvm-codegen` it will tell you the failures, then use the update_llc_test_checks.py script to regenerate the files

I'll recommit again, I'm just now seeing this

Reply:
uploaded the regenerated test files after removing Select instruction